src/newrelic/collector/client_test.go

```diff
@@ -91,3 +96,65 @@ func TestUserAgent(t *testing.T) {
 		}
 	}
 }
+
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (fn roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return fn(r)
+}
+
+func TestExecuteWhenMaxPayloadSizeExceeded(t *testing.T) {
+	// call is defined in collector_test.go
+	call.MaxPayloadSize = 0
+	testedFn := fmt.Sprintf("client.Execute(payload: %v, MaxPayloadSize: %v)", actualData, call.MaxPayloadSize)
+	var wantResponseBody []byte = nil
+	wantErr := errors.New("payload size too large:")
+
+	client := clientImpl{
+		httpClient: &http.Client{
+			Transport: roundTripperFunc(func(r *http.Request) (*http.Response, error) {
+				// no http.Response because no HTTP request should be made
+				return nil, nil
+			}),
+		},
+	}
+
+	b, err := client.Execute(call)
+	if b != nil {
+		t.Errorf("%s, got [%v], want [%v]", testedFn, b, wantResponseBody)
+	} else if err == nil {
+		t.Errorf("%s, got [%v], want [%v]", testedFn, err, wantErr)
+	} else if !strings.HasPrefix(err.Error(), wantErr.Error()) {
+		t.Errorf("%s, got [%v], want [%v]", testedFn, err, wantErr)
+	}
+}
+
+func TestExecuteWhenMaxPayloadSizeNotExceeded(t *testing.T) {
+	// call is defined in collector_test.go
+	call.MaxPayloadSize = 100
```

Comment:
See above comment about `call` and `actualData`.

Reply:
I plan to repeat [this](https://github.com/newrelic/newrelic-php-agent/pull/453/files#r877294219) solution.